paasta_tools/cli/cmds/local_run.py

```diff
@@ -696,24 +692,28 @@ def local_makefile_present():
         sys.stderr.write("Local Makefile with 'cook-image' target detected. Assuming --build\n")
         return True
     else:
-        sys.stderr.write("No Makefile with 'cook-image' target detected. Assuming --pull\n")
         return False
 
 
 def paasta_local_run(args):
-    if args.pull or args.dry_run:
-        build = False
-    elif args.build:
-        build = True
-    else:
-        build = local_makefile_present()
+    if (args.pull is not None) and (args.build is not None):
```

Comment:
Isn't this already taken care of ?
https://docs.python.org/2.7/library/argparse.html#mutual-exclusion